features/messages/impl/src/main/kotlin/io/element/android/features/messages/impl/timeline/protection/TimelineProtectionPresenter.kt

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2024 New Vector Ltd.
+ *
+ * SPDX-License-Identifier: AGPL-3.0-only
+ * Please see LICENSE in the repository root for full details.
+ */
+
+package io.element.android.features.messages.impl.timeline.protection
+
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.derivedStateOf
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import io.element.android.libraries.architecture.Presenter
+import io.element.android.libraries.matrix.api.core.EventId
+import io.element.android.libraries.preferences.api.store.AppPreferencesStore
+import kotlinx.collections.immutable.toImmutableSet
+import javax.inject.Inject
+
+class TimelineProtectionPresenter @Inject constructor(
+    private val appPreferencesStore: AppPreferencesStore,
+) : Presenter<TimelineProtectionState> {
+    @Composable
+    override fun present(): TimelineProtectionState {
+        val hideContent by appPreferencesStore.doesHideImagesAndVideosFlow().collectAsState(initial = false)
+        var allowedEvents by remember { mutableStateOf<Set<EventId>>(setOf()) }
```

Comment:
Should this be an immutable Set? I'm not sure if it might affect recompositions in any way, probably not.

Reply:
I think it is simpler to leave it like that.